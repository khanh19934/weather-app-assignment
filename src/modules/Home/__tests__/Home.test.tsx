// @ts-nocheck
import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup, act, fireEvent } from '@testing-library/react';
import HomeContainer from '../Home.screen';
import HomeView from '../Home.view';

const defaultProps = {
  errorMessage: '',
  isSearching: false,
  isUserDeniedPermission: false,
  handleSubmitSearch: jest.fn,
  handleSearchLocation: jest.fn,
  handleToggleShowSearchBox: jest.fn,
  initialLoadingText: '',
  weatherResult: { city: { name: 'asd' }, list: [] },
  handleFormatTime: () => jest.fn(),
  showSearchBox: false,
};

describe('Test Snap shot', () => {
  it('render correctly', () => {
    const tree = renderer.create(<HomeContainer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Snapshot home view', () => {
  const tree = renderer.create(<HomeView {...defaultProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Test View Via Props', () => {
  test('Should Show getting location in the first time use app', async () => {
    const { getByText } = render(
      <HomeView
        {...{
          ...defaultProps,
          initialLoadingText: 'We are try to getting your location. Please grant permission ...',
        }}
      />,
    );
    expect(
      getByText('We are try to getting your location. Please grant permission ...'),
    ).toBeInTheDocument();
  });

  test('Should Show Fetching data via location', async () => {
    const { getByText } = render(
      <HomeView
        {...{
          ...defaultProps,
          initialLoadingText: 'Fetching the data depend on your location now. Please wait...',
        }}
      />,
    );
    expect(
      getByText('Fetching the data depend on your location now. Please wait...'),
    ).toBeInTheDocument();
  });

  test('Should Show input search box when click button Change Location', async () => {
    const { getByTestId } = render(<HomeView {...{ ...defaultProps, showSearchBox: true }} />);

    expect(getByTestId('searchBox')).toBeVisible();
  });

  test('Should Show overlay when search box', async () => {
    const { getByTestId } = render(
      <HomeView {...{ ...defaultProps, showSearchBox: true, isSearching: true }} />,
    );

    act(() => {
      fireEvent.keyDown(getByTestId('searchBox', { key: 'Enter', code: 'Enter' }));
    });

    expect(getByTestId('overlay')).toBeVisible();
  });
});

describe('Test Home container', () => {
  beforeEach(async () => {
    navigator.permissions = { query: jest.fn().mockResolvedValue };
    navigator.geolocation = {
      getCurrentPosition: jest.fn().mockResolvedValue({ latitude: 1, longitude: 1 }),
    };
  });
  afterEach(cleanup);

  test('Show loading for get location', async () => {
    const { getByText } = render(<HomeContainer />);
    await act(async () => {
      await navigator.permissions.query({ name: 'geolocation' });
    });

    expect(
      getByText('Fetching the data depend on your location now. Please wait...'),
    ).toBeInTheDocument();
  });
});
