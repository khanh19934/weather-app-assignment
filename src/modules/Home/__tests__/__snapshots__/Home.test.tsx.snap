// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="container-fluid home"
>
  <div
    className="main-content"
  >
    <div
      className="left-container"
    >
      <div
        className="bg-overlay"
      />
      <div
        className="today-info-container"
      >
        <span
          className="day-of-week"
        />
        <span
          className="date"
        />
        <div
          className="location-group"
        >
          <svg
            className="location-group-icon"
            color="#fff"
          >
            location.svg
          </svg>
          <span>
            asd
          </span>
        </div>
        <img
          alt="weather"
          className="weather-icon"
          src="https://openweathermap.org/img/wn/02d@2x.png"
        />
        <span
          className="main-temp"
        >
          0
           째
        </span>
        <span
          className="text-capitalize"
        >
          
        </span>
      </div>
    </div>
    <div
      className="right-container"
    >
      <div
        className="d-flex justify-content-between mb-2 info-item"
      >
        <span
          className="text-white"
        >
          HUMIDITY
        </span>
        <span
          className="text-white"
        >
          0%
        </span>
      </div>
      <div
        className="d-flex justify-content-between mb-2 info-item"
      >
        <span
          className="text-white"
        >
          WIND SPEED
        </span>
        <span
          className="text-white"
        >
          0 m/s
        </span>
      </div>
      <div
        className="d-flex justify-content-between mb-2 info-item"
      >
        <span
          className="text-white"
        >
          FEELS LIKE
        </span>
        <span
          className="text-white"
        >
          0 째
        </span>
      </div>
      <div
        className="d-flex justify-content-between mb-2 info-item"
      >
        <span
          className="text-white"
        >
          HIGHEST TEMP
        </span>
        <span
          className="text-white"
        >
          0 째
        </span>
      </div>
      <div
        className="d-flex justify-content-between mb-2 info-item"
      >
        <span
          className="text-white"
        >
          LOWEST TEMP
        </span>
        <span
          className="text-white"
        >
          0 째
        </span>
      </div>
      <div
        className="forecast-container"
      />
      <div
        className="input-container"
      >
        <button
          className="button-change-location"
          data-testid="submitButton"
          onClick={[Function]}
        >
          <svg
            id="locationIcon"
          >
            location.svg
          </svg>
          Change Location
        </button>
      </div>
    </div>
  </div>
  <div
    className="home__bottom-warning"
  />
</div>
`;

exports[`Test Snap shot render correctly 1`] = `
<div
  className="container-fluid home"
>
  <div
    className="initial-loading"
  >
    <span>
      We are try to getting your location. Please grant permission ...
    </span>
  </div>
  <div
    className="home__bottom-warning"
  />
</div>
`;
